File: Semana_3/aula05_04_22.py
grafo = []

def novoGrafo(n_vertice):
    N = n_vertice
    for i in range(0, N):
        linha = list()
        for j in range(0, N):
            linha.append(0)
        grafo.append(linha)

############## AULA 05/04 ##############
def diagonalPrincipal(grafo,n_vertices):
    A = grafo
    for i in range(0,n_vertices):
        print(A[i][i])

def mostraCantos(grafo,n_vertices):
    A = grafo
    print(A[0][0])
    print(A[n_vertices-1][0])
    print(A[0][n_vertices-1])
    print(A[n_vertices-1][n_vertices-1])


novoGrafo(5)
print("-="*10,"Mostra a diagonal principal da matriz","-="*10)
diagonalPrincipal(grafo,5)
print("-="*10,"Mostra os quatro cantos da matriz","-="*10)
mostraCantos(grafo,5)

File: Semana_3/aula05_04_22.c
int A[MAX_SIZE][MAX_SIZE];
// número de vértices
int n;

#define MAX_SIZE 

//Criação da estrutura do grafo
void novoGrafo(int N){
  n=N;
  for (int i=0; i<n; i++){
    for(int j=0; j<n; j++){
      A[i][j]=0;
    }
  }
}

// Criando a função que mostra a diagonal principal da matriz. OBS (aij = aji) 
// Exemplos de elementos da diagonal principal (a11,a22,a33,a44,aii,ajj), ou seja, onde o índice da coluna é igual ao da linha
void mostraDiagonal(){
    printf("------------DIAGONAL PRINCIAL DA MATRIZ-----------");
    printf("\n");
    for (int i = 0; i<n; i++){
        printf("%d", A[i][i]); 
    }
    printf("\n");
}

// Criando a Função que mostra apenas os quatro cantos da matriz
// Basicamente, serão sempre os elementos (a00,a0 n-1, an-1 0, an-1 n-1) sendo a sempre linha coluna a ordem.
// Obs. estamos trabalhando com matrizes quadradas, então sempre haverá 4 cantos.

void mostraCantos(){
    printf("---------- Os 4 Cantos da Matriz------------------ \n");
    
    // Mostrando o primeiro canto. linha 0 coluna 0
    printf("%d",A[0][0]);
    
    // Monstrando o elemento da linha 0 e coluna n -1
    printf("%d",A[0][n-1]);
    
    // Monstrando o elemento de linha n-1 e coluna 0
    printf("%d",A[n-1][0]);
    
    // Mostrando o elemento de linha n-1 e coluna n-1
    printf("%d \n",A[n-1][n-1]);
}